comp/core/autodiscovery/providers/config_reader.go

```diff
@@ -355,6 +364,11 @@ func GetIntegrationConfigFromFile(name, fpath string) (integration.Config, error
 		return conf, err
 	}
 
+	// Check for empty file and return special error if so
+	if len(yamlFile) == 0 {
```

Comment:
Would it make sense to change the `Configuration file contains no valid instances` error below instead? It already explicitly checks for empty configuration files. That way we would also avoid errors on files consisting entirely of whitespace, which is something that can happen with when generating files with a template I believe.

Reply:
There's a lot of variations that can lead to `no valid instances`, for example `instances: null`. For both those and for whitespace, I'm attempting to be as conservative and efficient here as possible.

Can definitely imagine a scenario where a linter forces a customer to have a newline in a file, but I think we can treat that as a feature request to expand the tolerance for "empty" here if and when that happens.

Reply:
Is there a reason to treat `instances: []` as an error?

But if we want to, we can distinguish between pure whitespace and an empty struct by unmarshaling into a pointer: https://go.dev/play/p/JPTKqS_eUmm (note the pointer-to-pointer argument in `Unmarshal`).

Minor concern though, so LGTM from AML.

Reply:
Mostly just an attempt to keep this change minimal. I think there's potentially a larger follow on discussion to be had around what the agent expects from config files. The stub pattern where integrations have to put a fake instance to be considered valid is definitely unintuitive.